python/bifrost/blocks/guppi_raw.py

```diff
@@ -59,7 +59,7 @@ def on_sequence(self, reader, sourcename):
         df_MHz = bw_MHz / nchan
         f0_MHz = cfreq_MHz - 0.5*(nchan-1)*df_MHz
         # Note: This will be negative if OBSBW is negative, which is correct
-        dt_s   = 1. / df_MHz / 1e6
+        dt_s   = np.abs(1. / df_MHz / 1e6)
```

Comment:
Hmm. I think keeping OBSFREQ as-is for now is pragmatic. I'm happy with the ability to change step in my branch for now!